frontend/src/metabase/query_builder/components/view/ViewHeader/components/DashboardBackButton/DashboardBackButton.tsx

```diff
@@ -7,21 +7,28 @@ import * as Urls from "metabase/lib/urls";
 import { navigateBackToDashboard } from "metabase/query_builder/actions";
 import { getDashboard } from "metabase/query_builder/selectors";
 import { ActionIcon, type ActionIconProps, Icon, Tooltip } from "metabase/ui";
+import type { DashboardId } from "metabase-types/api";
 
 import DashboardBackButtonS from "./DashboardBackButton.module.css";
 
 export type DashboardBackButtonProps = {
   noLink?: boolean;
   onClick?: () => void;
+  dashboardOverride?: {
+    id: DashboardId;
+    name: string;
+  };
 } & ActionIconProps &
   HTMLAttributes<HTMLButtonElement>;
 
 export function DashboardBackButton({
   noLink,
   onClick,
+  dashboardOverride,
```

Comment:
Allow this component to override where to read the dashboard value from. In the SDK, we want to be able to pass the dashboard value rather than relying on Rudux.